autogen.pl

```diff
@@ -1397,42 +1397,87 @@ sub replace_config_sub_guess {
 # Make sure we got a submodule-full clone.  If not, abort and let a
 # human figure it out.
 if (-f ".gitmodules") {
-    open(IN, "git submodule status|")
-        || die "Can't run \"git submodule status\"";
-    while (<IN>) {
-        $_ =~ m/^(.)[0-9a-f]{40}\s+(\S+)/;
-        my $status = $1;
-        my $path   = $2;
-
-        print("=== Submodule: $path\n");
-        if (index($path, "pmix") != -1 and list_contains("pmix", @disabled_3rdparty_packages)) {
-          print("Disabled - skipping openpmix");
-          next;
-        }
-        if (index($path, "prrte") != -1 and list_contains("prrte", @disabled_3rdparty_packages)) {
-          print("Disabled - skipping prrte");
-          next;
+    if (! -d ".git") {
+        print("   Not in a git clone; unable to check git submodule status\n");
+        print("   Doing some sanity checks anyway...\n");
+
+        # Do a quick sanity check to ensure that non-3rd-party
+        # submodules are at least present (e.g., they won't be present
+        # if you downloaded a GitHub.com-created tarball).
+        open(IN, ".gitmodules") ||
+            die "Can't open .gitmodules";
+        while (<IN>) {
+            # Find "path = " lines
+            if (!($_ =~ m/^\s+path = (.+)$/)) {
+                next;
+            }
+            my $path = $1;
+
+            # Only care about paths that do not include "3rd-party"
```

Comment:
Errr...don't you mean "only care about paths that _do_ include..."? I believe that is what the conditional below is doing, but maybe I'm wrong?

Reply:
The conditional is: if it finds `3rd-party` in the path, it continues (`next`) to the next line in the file.